common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -89,7 +89,14 @@ private static void loadConfigOptions(List<ConfigHolder> holders) {
 
             // read value and update option
             JsonElement holderJson = configObject.get(holder.getJsonName());
-            Object value = gson.fromJson(holderJson, holder.getType());
+            Object value;
+
+            if (holder.getTypeOverride() == null) {
+                value = gson.fromJson(holderJson, holder.getType());
+            } else {
+                value = gson.fromJson(holderJson, holder.getTypeOverride());
+            }
+
```

Comment:
are you certain all of this (and the related type handling elsewhere) is necessary? i'm pretty sure gson is capable of automatically determining types in a map

Reply:
i did not find a way to make gson handle it properly and we do the same in WebManager.